contracts/libraries/UniswapRouter.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ISwapRouter} from "../interfaces/ISwapRouter.sol";
+import "./Path.sol";
+
+library UniswapRouter {
+    using Path for bytes;
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    function checkPath(
+        bytes memory swapPath,
+        address validTokenIn,
+        address validTokenOut
+    ) internal pure returns (bool isValidPath) {
+        (address tokenIn, address tokenOut, ) = swapPath.decodeFirstPool();
+
+        while (swapPath.hasMultiplePools()) {
+            swapPath = swapPath.skipToken();
+            (, tokenOut, ) = swapPath.decodeFirstPool();
+        }
+
+        return tokenIn == validTokenIn && tokenOut == validTokenOut;
+    }
+
+    function swap(
```

Comment:
Missing the function documentation comment

Reply:
Added documentation